staging/src/k8s.io/client-go/plugin/pkg/client/auth/oidc/oidc.go

```diff
@@ -185,6 +188,26 @@ func (p *oidcAuthProvider) Login() error {
 	return errors.New("not yet implemented")
 }
 
+// UpdateTokenAndCheckMatch takes a configuration map and updates the token of its cfg, and checks if it is match with the given config.
+func (provider *oidcAuthProvider) updateTokenAndCheckMatch(cfg map[string]string) bool {
+	provider.mu.Lock()
+	defer provider.mu.Unlock()
+	// if id-token and refresh-token in cfg, update provider.cfg
+	if cfg[cfgIDToken] != "" {
+		provider.cfg[cfgIDToken] = cfg[cfgIDToken]
```

Comment:
do we own the `cfg` map? I didn't expect this file to modify it

Reply:
Yes, the provider own the `cfg` map, If we don't modify here, the tokens may be invalid. The func `RoundTrip` rely on valid tokens. See [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/plugin/pkg/client/auth/oidc/oidc.go#L224).

And you can see [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/plugin/pkg/client/auth/oidc/oidc.go#L278), current code modify the `cfg` if we refresh the tokens.